### PR TITLE
Sql introspection

### PR DESCRIPTION
**Pull request by slinkydeveloper**

### CODE REVIEW

File: docs/operate/introspection.mdx [hunk 1]
```diff
@@ -73,265 +82,234 @@ The CLI only allows you to interact with active invocations. Active invocations
...
+```sql
+select * from sys_journal where id = '<INVOCATION_ID>';
+```
 
-        ```sql
-        select * from sys_journal where service = 'test.MyServiceName' and service_key = (
-        select service_key from sys_status where id = '<INVOCATION_ID>'
-        );
-        ```
+If you don't know the invocation id of your request, you can look it in the `sys_invocation` table as described [above](#listing-ongoing-invocations)
```
Comment (gvdongen) [inline docs/operate/introspection.mdx:146]:
```suggestion
If you don't know the invocation id of your request, you can find it in the `sys_invocation` table as described [above](#listing-ongoing-invocations)
```

File: docs/references/sql-introspection.mdx [hunk 2]
```diff
@@ -13,87 +13,98 @@ The schema of these tables is described in the following.
...
+| Column name       | Type     | Description                                                                                                                                                                    | Example   |
+|-------------------|----------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------|
+| `partition_key`   | `UInt64` | Internal column that is used for partitioning the services invocations. Can be ignored.                                                                                        |           |
+| `service_name`    | `Utf8`   | The name for the invoked service.                                                                                                                                              | `Greeter` |
+| `service_key`     | `Utf8`   | The service key.                                                                                                                                                               | `bob`     |
```
[the remark this answers is not in the document]
Comment (gvdongen) [reply docs/references/sql-introspection.mdx:82]: What is this value for plain services? Maybe mention this?

File: docs/references/sql-introspection.mdx [hunk 3]
```diff
@@ -13,87 +13,98 @@ The schema of these tables is described in the following.
...
+
+## Table: `sys_idempotency`
+
+| Column name       | Type     | Description                                                                                                                                                                    | Example             |
+|-------------------|----------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------|
+| `partition_key`   | `UInt64` | Internal column that is used for partitioning the services invocations. Can be ignored.                                                                                        |                     |
+| `service_name`    | `Utf8`   | The name for the invoked service.                                                                                                                                              | `Greeter`           |
+| `service_key`     | `Utf8`   | The service key.                                                                                                                                                               | `bob`               |
```
Comment (gvdongen) [inline docs/references/sql-introspection.mdx:93]:
```suggestion
| `service_key`     | `Utf8`   | The key of the Virtual Object.                                                                                                                                                               | `bob`               |
```

File: docs/references/sql-introspection.mdx [hunk 4]
```diff
@@ -13,87 +13,98 @@ The schema of these tables is described in the following.
...
+| `service_handler` | `Utf8`   | The invoked RPC method.                                                                                                                                                        | `greet`             |
```
Comment (gvdongen) [inline docs/references/sql-introspection.mdx:94]:
```suggestion
| `service_handler` | `Utf8`   | The invoked handler.                                                                                                                                                        | `greet`             |
```

**Review by gvdongen — APPROVED** [review]
LGTM after comments have been fixed :+1:

File: docs/operate/introspection.mdx [hunk 5]
```diff
@@ -28,15 +28,15 @@ Have a look at the [installation guide](/operate/cli) for the CLI.
 
 
 Restate exposes the following SQL tables, queryable via the [*psql* client](https://www.postgresql.org/docs/current/app-psql.html):
-- `sys_status` table to inspect ongoing invocation status
-- `sys_inbox` table to inspect pending invocation status
-- `sys_invocation_state` table to inspect the in-memory state of the invoker
-- `sys_journal` table to inspect the journal of running invocations
+- `sys_invocation` table to inspect invocations
+- `sys_inbox` table to inspect queue of pending invocations
+- `sys_keyed_service_status` table to inspect keyed services (e.g. virtual objects) status
```
[the remark this answers is not in the document]
Comment (slinkydeveloper) [reply docs/operate/introspection.mdx:33]: In future this table will show for workflows too. I'm ok with changing this now though, until we have the workflows

File: docs/references/sql-introspection.mdx [hunk 6]
```diff
@@ -13,87 +13,98 @@ The schema of these tables is described in the following.
...
+| `invoked_id`      | `Utf8`    | If this entry represents an outbound invocation, indicates the ID of that invocation.                                                                                                                  |                                              |
+| `invoked_target`  | `Utf8`    | If this entry represents an outbound invocation, indicates the invocation Target. Format for regular services: `SvcName/HandlerName`. Format for virtual objects: `VirtualObjectName/Key/HandlerName`. | `Greeter/greet` or `Greeter/Francesco/greet` |
+| `sleep_wakeup_at` | `Date64`  | If this entry represents a sleep, indicates wakeup time.                                                                                                                                               |                                              |
+| `raw`             | `Binary`  | Raw binary representation of the entry. Check the [service protocol](https://github.com/restatedev/service-protocol) for more details to decode it.                                                    |                                              |
+
+## Table: `sys_keyed_service_status`
```
[the remark this answers is not in the document]
Comment (slinkydeveloper) [reply docs/references/sql-introspection.mdx:67]: No, keyed services is an internal categorization of services, as opposed to unkeyed services. keyed services are virtual objects, but are also workflows. unkeyed services are "regular services", but also free floating functions in future.

File: docs/references/sql-introspection.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (slinkydeveloper) [reply docs/references/sql-introspection.mdx:82]: There is none

**Pull request merged**